x-pack/plugins/security_solution/public/alerts/components/alerts_table/helpers.ts

```diff
@@ -135,30 +144,64 @@ export const replaceTemplateFieldFromMatchFilters = (filters: Filter[], ecsData:
 
 export const reformatDataProviderWithNewValue = <T extends DataProvider | DataProvidersAnd>(
   dataProvider: T,
-  ecsData: Ecs
+  ecsData: Ecs,
+  timelineType: TimelineType = TimelineType.default
 ): T => {
-  if (templateFields.includes(dataProvider.queryMatch.field)) {
-    const newValue = getStringArray(dataProvider.queryMatch.field, ecsData);
-    if (newValue.length) {
+  // Support for legacy "template-like" timeline behavior that is using hardcoded list of templateFields
+  if (timelineType === TimelineType.default) {
+    if (templateFields.includes(dataProvider.queryMatch.field)) {
+      const newValue = getStringArray(dataProvider.queryMatch.field, ecsData);
+      if (newValue.length) {
+        dataProvider.id = dataProvider.id.replace(dataProvider.name, newValue[0]);
+        dataProvider.name = newValue[0];
+        dataProvider.queryMatch.value = newValue[0];
+        dataProvider.queryMatch.displayField = undefined;
+        dataProvider.queryMatch.displayValue = undefined;
+      }
+    }
+    dataProvider.type = dataProvider.type ?? DataProviderType.default;
```

Comment:
if what I am saying makes sense I think we can simplify this function, I do not think that we want to leave this function with dataProvider type to be a template. is that correct?